src/main/java/cms/gov/madie/measure/utils/JsonUtil.java

```diff
@@ -632,4 +644,53 @@ public static String getTestCaseJson(String json) throws JsonProcessingException
     }
     return null;
   }
+
+  // going through JsonNode, converts any datetime string into UTC datetime
+  public static JsonNode replaceNestedDateTimeStringValue(JsonNode node) {
+    if (node.isObject()) {
+      ObjectNode objectNode = (ObjectNode) node;
+      objectNode
+          .fields()
+          .forEachRemaining(
+              entry -> {
+                String fieldName = entry.getKey();
+                JsonNode childNode = entry.getValue();
+                String currentValue = childNode.asText();
+                String newValue = getNewValue(currentValue);
+                if (childNode.isTextual()) {
+                  objectNode.put(fieldName, newValue);
+                } else {
+                  replaceNestedDateTimeStringValue(childNode);
+                }
+              });
+    } else if (node.isArray()) {
+      node.forEach(childNode -> replaceNestedDateTimeStringValue(childNode));
+    }
+    return node;
+  }
+
+  // converts a value into UTC datetime, if the the passed in value is a datetime
+  // otherwise, returns the original value
+  static String getNewValue(String value) {
+    String newValue = value;
+    if (value.length() > 19 && (PATTERN.matcher(value.substring(0, 19)).matches())) {
```

Comment:
Why are we checking for a length of 19 or greater ?

Reply:
dateTime = ZonedDateTime.parse(value, FORMATTER); already checks to see if a value is of type DateTime, if not it throws DateTimeParseExcepttion, Do you think we even need this check ?

Reply:
@RohitKandimalla I think it's better to check if it's a datetime string, because most of the nodes are not, we don't even need to process non dateime nodes

Reply:
Yeah we don't have to but if we are already using ZonedDateTime.parse that will take care of pattern matching

Reply:
@RohitKandimalla Yeah I know, my original implementation didn't have this check, but since there are so many more nodes that are not datetime, I felt it's not necessary for them to go to catch block.

Reply:
@RohitKandimalla ok if you insist, I can remove it.

Reply:
if possible we should avoid regex